### PR TITLE
[RPA-1889] Fixes issue where accepted/deactivated user always landed in default organization

### PR DESCRIPTION
**Pull request by bhszewczyk**
## Description

- Accepting user who has registered via link now lands under proper tenant
- Deactivated user not lands under proper tenant

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

---

I agree to the terms of the RunBotics Contributor License Agreement.

### CODE REVIEW

File: runbotics/runbotics-scheduler/src/scheduler-database/user/user.controller.ts [hunk 1]
```diff
@@ -42,7 +42,7 @@ export class UserController {
         @Body(new ZodValidationPipe(updateUserSchema)) userDto: UpdateUserDto
     ) {
         return this.userService.update(
-            { ...userDto, tenantId: null },
```
Comment (marcin-depcik) [inline runbotics/runbotics-scheduler/src/scheduler-database/user/user.controller.ts:45]: this was intentional, if it is necessary to change, please adjust logic of `.update()`

**Review by marcin-depcik — APPROVED** [review]

File: runbotics/runbotics-orchestrator-ui/src/src-app/views/users/UsersListView/UsersListView.tsx [hunk 2]
```diff
@@ -62,7 +62,7 @@ const UsersListView: FC = () => {
     };
 
     useEffect(() => {
-        dispatch(tenantsActions.getAll());
+        hasAdminAccess && dispatch(tenantsActions.getAll());
```
Comment (star-in-the-jar) [inline runbotics/runbotics-orchestrator-ui/src/src-app/views/users/UsersListView/UsersListView.tsx:65]: thinking of the future I think it's good to rename booleans like that to hasCrossTenantAccess or sth

Comment (star-in-the-jar) [reply runbotics/runbotics-orchestrator-ui/src/src-app/views/users/UsersListView/UsersListView.tsx:65]: cause now we have two admins

Comment (star-in-the-jar) [reply runbotics/runbotics-orchestrator-ui/src/src-app/views/users/UsersListView/UsersListView.tsx:65]: why add this condition if TENANT_ADMIN has access to this tab too? The difference is that TENANT_ADMIN sees users in his/hers tenant, and ADMIN sees all users across all tenants.

But the refresh should happen for both of them, or am I wrong?

File: runbotics/runbotics-orchestrator-ui/src/src-app/views/users/UsersRegistrationView/UsersRegistration.utils.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import { UserDto } from 'runbotics-common';
+
+export const getSelectedTenants = (users: UserDto[]) => {
```
Comment (star-in-the-jar) [inline runbotics/runbotics-orchestrator-ui/src/src-app/views/users/UsersRegistrationView/UsersRegistration.utils.ts]: add return type please

Comment (star-in-the-jar) [reply runbotics/runbotics-orchestrator-ui/src/src-app/views/users/UsersRegistrationView/UsersRegistration.utils.ts]: If it's an object `{
<userId>: <tenantId>
}`

why is the function name getSelectedTenants?

File: runbotics/runbotics-orchestrator-ui/src/src-app/views/users/UsersRegistrationView/UsersRegistrationView.tsx [hunk 4]
```diff
@@ -137,6 +138,9 @@ const UsersRegistrationView: FC = () => {
 
     useEffect(() => {
         const allUsers = hasAdminAccess ? notActivated.allByPage : tenantNotActivated.allByPage;
+        setSelectedTenants(hasAdminAccess
+            ? getSelectedTenants(notActivated.allByPage?.content)
+            : getSelectedTenants(tenantNotActivated.allByPage?.content));
```
Comment (star-in-the-jar) [inline runbotics/runbotics-orchestrator-ui/src/src-app/views/users/UsersRegistrationView/UsersRegistrationView.tsx]: setSelectedTenants(
    hasAdminAccess
              ? getSelectedTenants(notActivated.allByPage?.content)
              : getSelectedTenants(tenantNotActivated.allByPage?.content)
);

Comment (star-in-the-jar) [reply runbotics/runbotics-orchestrator-ui/src/src-app/views/users/UsersRegistrationView/UsersRegistrationView.tsx]: formatting matter

File: runbotics/runbotics-orchestrator-ui/src/src-app/views/users/UsersRegistrationView/UsersRegistrationView.tsx [hunk 5]
```diff
@@ -156,11 +160,11 @@ const UsersRegistrationView: FC = () => {
 
     return (
         <>
-            <DeleteUserDialog
+            {hasAdminAccess && <DeleteUserDialog
```
Comment (star-in-the-jar) [inline runbotics/runbotics-orchestrator-ui/src/src-app/views/users/UsersRegistrationView/UsersRegistrationView.tsx:163]: again, wy to add this if TENANT_ADMIN can delete users too? (as far as I know he/she should be able to)

**Review by star-in-the-jar — CHANGES_REQUESTED** [review]

File: runbotics/runbotics-scheduler/src/scheduler-database/user/user.controller.ts (hunk 1 above)
Comment (bhszewczyk) [reply runbotics/runbotics-scheduler/src/scheduler-database/user/user.controller.ts:45]: no need as tenantId is part of userDto

File: runbotics/runbotics-orchestrator-ui/src/src-app/views/users/UsersListView/UsersListView.tsx (hunk 2 above)
Comment (bhszewczyk) [reply runbotics/runbotics-orchestrator-ui/src/src-app/views/users/UsersListView/UsersListView.tsx:65]: It has, but this endpoint is for retrieving all tenants. in Users tab tenant admin do not see the list of tenants to choose from and in the table there i no tenant column (as all already is within the tenant)

File: runbotics/runbotics-orchestrator-ui/src/src-app/views/users/UsersRegistrationView/UsersRegistration.utils.ts (hunk 3 above)
Comment (bhszewczyk) [reply runbotics/runbotics-orchestrator-ui/src/src-app/views/users/UsersRegistrationView/UsersRegistration.utils.ts]: to be consistent with what was defined previously in the UsersRegistartionView. do you have any suggestion to change it?
image

File: runbotics/runbotics-orchestrator-ui/src/src-app/views/users/UsersRegistrationView/UsersRegistrationView.tsx (hunk 5 above)
Comment (bhszewczyk) [reply runbotics/runbotics-orchestrator-ui/src/src-app/views/users/UsersRegistrationView/UsersRegistrationView.tsx:163]: No, tenant admin cannot delete users. tenant admin can DEACTIVATE but cannot DELETE

File: runbotics/runbotics-orchestrator-ui/src/src-app/views/users/UsersRegistrationView/UsersRegistration.utils.ts (hunk 3 above)
Comment (star-in-the-jar) [reply runbotics/runbotics-orchestrator-ui/src/src-app/views/users/UsersRegistrationView/UsersRegistration.utils.ts]: it returns `{ <userId>: <tenantId> }` so I suggest the name - combineUserWithTenantIds or another one that describes what the function do

Also current useState name is a bit confusingf

**Review by star-in-the-jar — APPROVED** [review]

**Pull request merged**